realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -334,6 +343,44 @@ private void emitAccessors(JavaWriter writer) throws IOException {
         }
     }
 
+    private void emitCodeForInjectingObjectContext(JavaWriter writer, boolean isSetter) throws IOException {
+        // if invoked from model's constructor, inject BaseRealm and Row
+        writer.beginControlFlow("if (proxyState == null)");
+        {
+            writer.emitSingleLineComment("Called from model's constructor. Inject context.");
+            writer.emitStatement("injectObjectContext()");
+        }
+        writer.endControlFlow();
+        writer.emitEmptyLine();
+
+        if (isSetter) {
+            writer.beginControlFlow(
+                    "if (proxyState.isUnderConstruction() && !proxyState.getRealm$realm().isInTransaction())")
```

Comment:
`isInTransaction` is a pretty expensive call. I am not sure it will be acceptable to call that for all accessors.